### PR TITLE
Fix several clippy warnings in components/bluetooth

### PR DESCRIPTION
**Pull request by mnaibei**
Fixes several clippy warnings in the components/bluetooth directory. It is part of https://github.com/servo/servo/issues/31500.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #31500

- [x] These changes do not require tests because they do not modify functionality, they only fix lint errors.

### CODE REVIEW

File: components/bluetooth/lib.rs [hunk 1]
```diff
@@ -120,20 +120,20 @@ fn matches_filter(device: &BluetoothDevice, filter: &BluetoothScanfilter) -> boo
     if !filter.get_services().is_empty() {
         if let Ok(device_uuids) = device.get_uuids() {
             for service in filter.get_services() {
-                if device_uuids.iter().find(|x| x == &service).is_none() {
+                if device_uuids.iter().any(|x| &x == &service) {
```
Comment (mrobinson) [inline components/bluetooth/lib.rs:123]: I think this reverses the meaning of this if-statement. In the previous version it was checking if there were no results from the find and in this version it is true if there are any.

**Review by mrobinson — CHANGES_REQUESTED** [review]

File: components/hyper_serde/lib.rs [hunk 2]
```diff
@@ -138,7 +138,7 @@ impl<T> De<T> {
     /// Returns a new `De` wrapper
     #[inline(always)]
     pub fn new(v: T) -> Self {
-        De { v: v }
+        De { v }
```
Comment (mrobinson) [inline components/hyper_serde/lib.rs:141]: It looks like you have also included your changes from your other PR. Please remove these changes from the PR.

File: components/bluetooth/lib.rs (hunk 1 above)
Comment (mnaibei) [reply components/bluetooth/lib.rs]: @mrobinson I have updated this, please have a look

**Review by mrobinson — APPROVED** [review]
Thanks! This looks great!

**Pull request merged**